autopart-hibernation.ks.in

```diff
@@ -0,0 +1,77 @@
+#test name: autopart-nohome
+%ksappend repos/default.ks
+network --bootproto=dhcp
+
+bootloader --timeout=1
+zerombr
+clearpart --all --initlabel
+
+# Include the generated autopart command.
+%include /tmp/ksinclude
+
+keyboard us
+lang en
+timezone America/New_York
+rootpw qweqwe
+shutdown
+
+%packages
+%end
+
+%pre
+# Generate the autopart command.
+autopart="autopart"
+if [ "@KSTEST_OS_NAME@" == "fedora" ]; then
+    # --hibernation should work even with --noswap
+    autopart="$autopart --noswap"
+fi
+echo "$autopart --hibernation" >/tmp/ksinclude
+%end
+
+
+%post
+err=0
+
+# Test if Grub is configured to resume from sleep.
+grub_path="/etc/default/grub"
+grub_param="GRUB_CMDLINE_LINUX"
+cmdline="$(grep $grub_param $grub_path | grep 'resume')"
+
+if [ -z "$cmdline" ]; then
+    echo "*** \"resume\" parameter is missing from $grub_param in $grub_path" >> /root/RESULT
+    err=1
+fi
+
+
+# Test if swap is equal or greater than ram size
+# Size of total memory in bytes
+mem_size="$(free -b | grep 'Mem:' | sed 's/^[^ ]*[ ]*//' | sed 's/[ ].*$//')"
+# Path to swap partition
+swap_device="$(blkid | grep 'TYPE="swap"' | grep -v "zram" | cut -d ':' -f 1)"
+
+if [ -z $swap_device ]; then
+    echo "*** SWAP partition was not created" >> /root/RESULT
+    err=1
+elif [ -z "$(grep $swap_device /etc/fstab)" ]; then
+    echo "*** SWAP is missing from fstab" >> /root/RESULT
+    err=1
+fi
+
+
+# Mount swap so we can accurately measure it's size.
+swapon "$swap_device"
+# Size of swap in bytes
+swap_size=$(swapon --raw --b | sed 1d | grep -v "zram" | cut -d ' ' -f 3)
+
+# Check if swap is large enough for current memeory configuration
+if [ "$swap_size" ] && [ "$(expr "$swap_size")" -le "$mem_size" ]; then
```

Comment:
The test should fail if `$swap_size` is an empty string.

Do you really need `expr`?

Reply:
Good point.

Yes I forgot to remove the `expr`.
Thanks.